✨ 프로그램 목록 App Icon 추가

padosum opened the pull request:
#26 
- extract-file-icon 설치
- icon이 저장되는 경로: /public/assets/images
- icon이 이미 저장되었는지 체크 후 저장
- ProgramTable 컴포넌트에서 bundleId 이름으로 저장된 이미지 로드

padosum committed:
✨ 프로그램 목록 App Icon 추가

- extract-file-icon 설치
- icon이 저장되는 경로: /public/assets/images
- icon이 이미 저장되었는지 체크 후 저장
- ProgramTable 컴포넌트에서 bundleId 이름으로 저장된 이미지 로드

the1020 commented on public/electron.js:
```diff
@@ -29,6 +30,9 @@ app.whenReady().then(() => {
     const activeWin = require('active-win');
     let window = await activeWin({ screenRecordingPermission: true });
 
+    const bundleId = window.owner.bundleId.replace(/\./g, '-');
```
window.owner.bundleId는 macOS에서만 받아올 수 있어서 window.owner.name을 사용해야 합니다.

the1020 commented on public/utils/icon.js:
```diff
@@ -0,0 +1,22 @@
+};
+
+const storeAppIcon = (appName, filePath) => {
+  const pathToAsset = path.join(__dirname, `../assets/images/${appName}.png`);
```
`/assets/images` 경로가 없으면 파일이 저장되지 않습니다.
=> 해당 경로가 있는지 확인해서 없으면 만들어주고 저장소에도 해당 경로 올려두기

padosum replied:
`.gitignore` 파일에 해당 경로 추가하기

padosum replied:
> `/assets/images` 경로가 없으면 파일이 저장되지 않습니다. => 해당 경로가 있는지 확인해서 없으면 만들어주고 저장소에도 해당 경로 올려두기

[이미지](https://freesvg.org/ios-7-icon-template)

padosum replied:
> `/assets/images` 경로가 없으면 파일이 저장되지 않습니다. => 해당 경로가 있는지 확인해서 없으면 만들어주고 저장소에도 해당 경로 올려두기

그냥 저장소에 해당 경로를 추가해두었음!

the1020 commented on src/components/ProgramTable.js:
```diff
@@ -170,7 +171,20 @@ export default function ProgramTable(props) {
+                        }}>
+                        <img
+                          style={{ width: '20px' }}
+                          src={`../../assets/images/${row.bundleId}.png`}
```
이미지 경로가 없는 경우 대체 이미지 넣기

padosum replied:
Name column에 아이콘 넣기

padosum replied:
> 이미지 경로가 없는 경우 대체 이미지 넣기

[이미지](https://freesvg.org/ios-7-icon-template)

the1020 commented on src/components/ProgramTable.js:
```diff
@@ -128,17 +129,17 @@ EnhancedTableHead.propTypes = {
       nextId(),
       name,
       title,
       format(startDate, 'HH:mm'),
```
date 포맷을 `HH:mm:ss`로 수정하기

the1020 reviewed (changes requested):
빠른 시일 내에 수정해주세요

padosum committed:
💄 시간 포맷 변경

- HH:mm -> HH:mm:ss

padosum committed:
🙈 .gitignore 업데이트

- .DS_Store 제외
- public/assets/images 디렉토리 no-icon.png 외 이미지 파일 제외

padosum committed:
🍱 app icon 대체 이미지 파일 추가

padosum committed:
♻️ app icon 추가 관련 수정

- app icon 이미지 경로에 파일이 없는 경우 대체 이미지 설정
- 'Name' 컬럼에 icon 추가하기
- icon 저장 파일명은 으로 변경
